### DISCUSSION TITLE
Kafka Connect

### DISCUSSION THREAD

**Discussion by Abdul1929** [strimzi/strimzi-kafka-operator#5885] [Q&A]
Hi All,

I have kafka and kafka connect single replica deployment with the helm chart.. need to know, how to perform single cluster rolling update in kafka connect..

**Comment by Abdul1929** [comment]
Anyone pls suggest here.. how to proceed with rolling update.. what steps need to perform

**Comment by Abdul1929** [comment]
> Hi All,
> 
> I have kafka and kafka connect single replica deployment with the helm chart.. need to know, how to perform single cluster rolling update in kafka connect..

@scholzj Can you suggest the steps

**Reply by scholzj** [reply]
When you change the Kafka or KafkaConnect custom resources, the operator will roll them automatically. You do not need to do anything.

**Reply by Abdul1929** [reply]
> When you change the Kafka or KafkaConnect custom resources, the operator will roll them automatically. You do not need to do anything.

Thanks for the reply.. I was checking the official strimzi doc for 0.26.0 but didn't get the example like what changes need to make in custom resources

**Reply by scholzj** [reply]
The operator has the logic inside of what requires rolling update and what doesn't require it. Most things do, but for example some Kafka configuration options can be set dynamically.

**Reply by Abdul1929** [reply]
okay.. can we have any doc or example for dynamic kafka configuration for this

**Reply by scholzj** [reply]
The Kafka docs list which options are configurable dynamically and which are not.

**Reply by Abdul1929** [reply]
okay I will check.. thanks

**Reply by Abdul1929** [reply]
No related to rolling update but regarding my current deployment.
Having kafka deployment so I had enabled metrics on that and tested with load on that.. like created topics and produces thousand of msgs but in that process kafka pod not working.. Its a test environment so I have tried to redeploy again but pod not coming up.
below is the error giving.

2021-11-13 17:32:10,777 WARN [ReplicaManager broker=0] Broker 0 stopped fetcher for partitions  and stopped moving logs for partitions  because they are in the failed log directory /var/lib/kafka/data/kafka-log0. (kafka.server.ReplicaManager) [LogDirFailureHandler]
2021-11-13 17:32:10,777 WARN Stopping serving logs in dir /var/lib/kafka/data/kafka-log0 (kafka.log.LogManager) [LogDirFailureHandler]
2021-11-13 17:32:10,780 **ERROR Shutdown broker because all log dirs in /var/lib/kafka/data/kafka-log0 have failed** (kafka.log.LogManager) [LogDirFailureHandler]

there is no such path (/var/lib/kafka/data/) for the logs in kafka deployment..

**Reply by Abdul1929** [reply]
added this property in kafka config and redeploy but same issue
auto.create.topics.enable: "false"

**Reply by scholzj** [reply]
`Shutdown broker because all log dirs in /var/lib/kafka/data/kafka-log0 have failed` ... it suggests some problem with your storage.

**Reply by Abdul1929** [reply]
Okay but with same storage I can able to deploy the kafka cluster with different namespace..

**Reply by scholzj** [reply]
What storage do you use? Each broker node has its own storage. So just because the disk for one broker is broken, it does not mean other disk for other broker is broken.

**Reply by Abdul1929** [reply]
okay got your point now.. so how to fix this Shutdown broker issue... any suggestion how to proceed

**Reply by scholzj** [reply]
Well, you will need to figure out what is the problem with the storage.
